ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/metacard-archive/presentation.tsx

```diff
@@ -27,92 +33,58 @@ type Props = {
 
 const render = (props: Props) => {
   const { onArchiveConfirm, onRestoreConfirm, isDeleted, loading } = props
-  const archiveDialogState = useDialogState()
-  const restoreDialogState = useDialogState()
-  return !loading ? (
+  const addSnack = useSnack()
+  const dialogContext = useDialog()
+  return (
     <>
-      <archiveDialogState.MuiDialogComponents.Dialog
-        {...archiveDialogState.MuiDialogProps}
-      >
-        <archiveDialogState.MuiDialogComponents.DialogTitle>
-          Are you sure you want to delete?
-          <div>
-            Doing so will remove the item(s) from future search results.
-          </div>
-        </archiveDialogState.MuiDialogComponents.DialogTitle>
-        <archiveDialogState.MuiDialogComponents.DialogActions>
-          <Button
-            onClick={() => {
-              archiveDialogState.handleClose()
-            }}
-          >
-            Cancel
-          </Button>
-          <Button
-            onClick={() => {
-              archiveDialogState.handleClose()
-              onArchiveConfirm()
-            }}
-          >
-            Delete
-          </Button>
-        </archiveDialogState.MuiDialogComponents.DialogActions>
-      </archiveDialogState.MuiDialogComponents.Dialog>
-      <archiveDialogState.MuiDialogComponents.Dialog
-        {...restoreDialogState.MuiDialogProps}
-      >
-        <archiveDialogState.MuiDialogComponents.DialogTitle>
-          Are you sure you want to restore? Doing so will include the item(s) in
-          future search results.
-        </archiveDialogState.MuiDialogComponents.DialogTitle>
-        <archiveDialogState.MuiDialogComponents.DialogActions>
-          <Button
-            onClick={() => {
-              restoreDialogState.handleClose()
-            }}
-          >
-            Cancel
-          </Button>
-          <Button
-            onClick={() => {
-              restoreDialogState.handleClose()
-              onRestoreConfirm()
-            }}
-          >
-            Restore
-          </Button>
-        </archiveDialogState.MuiDialogComponents.DialogActions>
-      </archiveDialogState.MuiDialogComponents.Dialog>
-      {!isDeleted ? (
+      <DialogTitle>
+        Are you sure you want to {isDeleted ? 'restore' : 'delete'}?
+      </DialogTitle>
+      <DialogContent>
+        <DialogContentText>
+          Are you sure you want to {isDeleted ? 'restore' : 'delete'}?
+        </DialogContentText>
+        <DialogContentText>
+          Doing so will {isDeleted ? 'add' : 'remove'} the item(s) from future
+          search results.
```

Comment:
a little nit picky, but:

```suggestion
          Doing so will {isDeleted ? 'include' : 'remove'} the item(s) {isDeleted ? 'in' : 'from'} future
          search results.
```